identity-service/src/notifications/index.js

```diff
@@ -281,6 +353,57 @@ class NotificationProcessor {
     }
     return metadata.max_block_number
   }
+
+  /**
+   * Doing the solana notification things
+   * @param {AudiusLibs} audiusLibs
+   * @param {number} minSlot min slot number to start querying discprov for new notifications
+   * @param {number} oldMaxSlot last max slot number seen
+   */
+  async indexAllSolanaNotifications (audiusLibs, minSlot, oldMaxSlot) {
+    const startDate = Date.now()
+    const startTime = process.hrtime()
+    const logLabel = 'notifications main indexAllSolanaNotifications job'
+
+    logger.info(`${logLabel} - minSlot: ${minSlot}, oldMaxSlot: ${oldMaxSlot}, startDate: ${startDate}, startTime: ${startTime}`)
+
+    // const { discoveryProvider } = audiusLibsWrapper.getAudiusLibs()
+    const { discoveryProvider } = audiusLibs
+
+    // Timeout of 2 minutes
+    const timeout = 2 /* min */ * 60 /* sec */ * 1000 /* ms */
+    const { info: metadata, notifications } = await discoveryProvider.getSolanaNotifications(minSlot, timeout)
+    logger.info(`${logLabel} - query solana notifications from discovery node complete`)
+
+    // Use a single transaction
+    const tx = await models.sequelize.transaction()
+    try {
+      // Insert the solana notifications into the DB
+      await processNotifications(notifications, tx)
+      logger.info(`notifications main indexAll job - processNotifications complete`)
+
+      // Fetch additional metadata from DP, query for the user's notification settings, and send push notifications (mobile/browser)
+      await sendNotifications(audiusLibs, notifications, tx)
+      logger.info(`notifications main indexAll job - sendNotifications complete`)
+
+      // Commit
+      await tx.commit()
+
+      // actually send out push notifications
+      await drainPublishedMessages()
+      logger.info(`notifications main indexAll job - drainPublishedMessages complete`)
```

Comment:
I'm not the biggest fan of the push and drain of published messages, but I think this should be broken out for solana notifications - else this method is triggered by both the regular notifs indexing and solana notifs indexing.